A copy of this since it is no longer exported by the kernel headers
--- a/vmblock-only/shared/compat_namei.h	2015-02-07 03:11:55.000000000 +0300
+++ c/vmblock-only/shared/compat_namei.h	2015-02-24 03:51:25.235286047 +0300
@@ -21,6 +21,20 @@
 
 #include <linux/namei.h>
 
+/* Copy-n-paste from kernel's source/fs/namei.c */
+struct nameidata {
+         struct path     path;
+         struct qstr     last;
+         struct path     root;
+         struct inode    *inode; /* path.dentry.d_inode */
+         unsigned int    flags;
+         unsigned        seq, m_seq;
+         int             last_type;
+         unsigned        depth;
+         struct file     *base;
+         char *saved_names[MAX_NESTED_LINKS + 1];
+};
+
 /*
  * In 2.6.25-rc2, dentry and mount objects were removed from the nameidata
  * struct. They were both replaced with a struct path.
